### ISSUE TITLE
OTLP GRPC exporters don't respect domain NO_PROXY settings due to grpc bug

### ISSUE THREAD

**Issue by akats7** [open-telemetry/opentelemetry-go#5976]
### Description

Brought up this [issue](https://github.com/open-telemetry/opentelemetry-collector/issues/11537) in the collector repo, realized it should likely be applied here as well. 

There's a bug in grpc-go ([here](https://github.com/grpc/grpc-go/issues/7779) and [here](https://github.com/grpc/grpc-go/issues/7556#issuecomment-2307484808)) related to the way NewClient does hostname resolution where now the host is resolved before the ProxyFromEnvironment function is called, due to this domain based no_proxy settings are not applied as the comparison is done only against the ip. localhost is not affected.

The grpc team is working on a fix but it seems like until that's in place we might want to revert to DialContext instead of NewClient.

### Environment

- Go Version: 1.23.2
- opentelemetry-go version: 1.32.0

### Steps To Reproduce

Set HTTP/S_PROXY
Init OTLP GRPC client with target endpoint
Include endpoint sub-domain in NO_PROXY

### Expected behavior

HTTP/S_PROXY setting should still be applied

**Comment by wasim-nihal** [comment]
Hello @MrAlias! I would like to contribute the fix for this issue.

**Comment by MrAlias** [comment]
Short term workaround: https://github.com/grpc/grpc-go/issues/7556#issuecomment-2307484808

We plan to hold off on this until the next release of grpc: https://github.com/grpc/grpc-go/milestone/67

If it is not resolved in that release we will re-evaluate applying a more permanent fix here.

**Comment by wasim-nihal** [comment]
sure @MrAlias

**Comment by MrAlias** [comment]
Note: the collector has already switched back to using the `Dial` function.

**Comment by akats7** [comment]
> Note: the collector has already switched back to using the `Dial` function.

Yep I made that change since the collector sanitizes the endpoint so we can't use the passthrough resolver

**Comment by MrAlias** [comment]
grpc 1.69.0 [release notes](https://github.com/grpc/grpc-go/releases/tag/v1.69.0#:~:text=The%20recently%20added%20grpc.NewClient%20function%20is%20incompatible%20with%20forward%20proxies%2C%20because%20it%20resolves%20the%20target%20hostname%20on%20the%20client%20instead%20of%20passing%20the%20hostname%20to%20the%20proxy.%20A%20fix%20is%20expected%20to%20be%20a%20part%20of%20grpc%2Dgo%20v1.70.%20(%237556)) mention this will be addressed in 1.70.0.

**Comment by MrAlias** [comment]
It looks like the fix has merged, but is postponed to the `v1.71.0` mileston.

https://github.com/grpc/grpc-go/pull/7881
